### PR TITLE
Clean up LayerManager context handling

### PR DESCRIPTION
**Pull request by ibgreen**
For #1421 
#### Background
- Our messy internal context handling interfered with the auto size feature.
#### Change List
- Copy only props that are actually wanted to the context object.
- Remove `oldContext`
- Remove unused props from

### CODE REVIEW

File: src/core/lib/layer-manager.js [hunk 1]
```diff
@@ -214,19 +230,23 @@ export default class LayerManager {
...
+      if (parameters.drawPickingColors !== this.drawPickingColors) {
+        this.drawPickingColors = parameters.drawPickingColors;
         this.setNeedsRedraw('drawPickingColors changed');
       }
     }
 
-    Object.assign(this.context, parameters);
+    // A way for apps to add data to context to access in layers
+    if ('context' in parameters) {
+      Object.assign(this.context.userData, parameters.context);
```
Comment (Pessimistress) [inline src/core/lib/layer-manager.js]: Does this mean that old `userData` fields may never go away?

File: src/core/lib/layer-manager.js [hunk 2]
```diff
@@ -491,17 +512,16 @@ export default class LayerManager {
     const viewportChanged = !oldViewport || !viewport.equals(oldViewport);
 
     if (viewportChanged) {
-      Object.assign(this.oldContext, this.context);
+      log.log(4, 'Viewport', viewport)();
+
       this.context.viewport = viewport;
       this.context.viewportChanged = true;
```
Comment (Pessimistress) [inline src/core/lib/layer-manager.js]: If `viewportChanged` is in `changeFlags`, do we still need this?

Comment (ibgreen) [reply src/core/lib/layer-manager.js]: Removed, nicely spotted!

File: src/core/lib/layer-manager.js (hunk 1 above)
Comment (ibgreen) [reply src/core/lib/layer-manager.js]: Good point. Updated the code to just link to the provided data object. This is a new field so no risk of breaking.

**Pull request merged**